sktime/forecasting/chronos.py

```diff
@@ -0,0 +1,268 @@
+"""Implements Chronos forecaster."""
+
+__author__ = ["Z-Fran", "benheid"]
+# __all__ = ["ChronosForecaster"]
+
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from skbase.utils.dependencies import _check_soft_dependencies
+
+from sktime.forecasting.base import ForecastingHorizon, _BaseGlobalForecaster
+from sktime.libs.chronos import ChronosPipeline
+
+if _check_soft_dependencies("torch", severity="none"):
+    import torch
+else:
+
+    class torch:
+        """Dummy class if torch is unavailable."""
+
+        bfloat16 = None
+
+
+if _check_soft_dependencies("transformers", severity="none"):
+    import transformers
+
+
+class ChronosForecaster(_BaseGlobalForecaster):
+    """Chronos forecaster.
+
+    Parameters
+    ----------
+    model_path : str
+        Path to the Chronos' huggingface model.
+    config : dict, default={}
+        Configuration to use for the model.
+    seed: int, optional, default=None
+        Random seed for transformers.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.chronos import ChronosForecaster
+    >>> from sktime.split import temporal_train_test_split
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> y = load_airline()
+    >>> y_train, y_test = temporal_train_test_split(y)
+    >>> fh = ForecastingHorizon(y_test.index, is_relative=False)
+    >>> forecaster = ChronosForecaster("amazon/chronos-t5-tiny")
+    >>> forecaster.fit(y_train)
+    >>> y_pred = forecaster.predict(fh)
+    """
+
+    # tag values are "safe defaults" which can usually be left as-is
+    _tags = {
+        "python_dependencies": ["torch", "transformers"],
+        "requires-fh-in-fit": False,
+        "X-y-must-have-same-index": True,
+        "enforce_index_type": None,
+        "handles-missing-data": False,
+        "capability:pred_int": False,
+        "X_inner_mtype": ["pd.DataFrame", "pd-multiindex", "pd_multiindex_hier"],
+        "y_inner_mtype": [
+            "pd.DataFrame",
+            "pd-multiindex",
+            "pd_multiindex_hier",
+        ],
+        "scitype:y": "univariate",
+        "capability:insample": False,
+        "capability:pred_int:insample": False,
+        "capability:global_forecasting": True,
+        "authors": ["Z-Fran", "benheid"],  # TODO original authors!,
+    }
+
+    _default_config = {
+        "num_samples": None,  # int, use value from pretrained model if None
+        "temperature": None,  # float, use value from pretrained model if None
+        "top_k": None,  # int, use value from pretrained model if None
+        "top_p": None,  # float, use value from pretrained model if None
+        "limit_prediction_length": False,  # bool
+        "torch_dtype": torch.bfloat16,  # torch.dtype
+        "device_map": "cpu",  # str, use "cpu" for CPU inference, "cuda" for gpu and "mps" for Apple Silicon # noqa
+    }
```

Comment:
This config is different from transformer's config right?

How do the users know how the config should look like? Since this in not the transformer's config, they cannot find this from huggingface

I would suggest
1. instead of keeping the config in python dict, put all the items as model parameters, that would add 7 more parameters during `__init__`
2. or inform users in the docstring how the config dict should look like

Reply:
(2) instructions on config have been added to the docstring